### PR TITLE
Add support for DBFS Blocks and Clusters

### PR DESCRIPTION
**Pull request by xinsnake**
This PR adds support for the operator to deploy DBFS files and Clusters from K8S.

### CODE REVIEW

**Comment by xinsnake** [comment]
DBFS is done, Cluster is not yet implemented, not ready to be merged.

**Comment by xinsnake** [comment]
@Azadehkhojandi   Please review the added capability of the operator: Cluster and DBFS.

Test Coverage | % | Lines Covered/Total
-- | -- | --
github.com/microsoft/azure-databricks-operator/api/v1 | 100.00% | 93/93
github.com/microsoft/azure-databricks-operator/controllers | 76.53% | 313/409
Report Total | 80.88% | 406/502

**Comment by xinsnake** [comment]
#29

File: api/v1/dcluster_types_extra.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	LastStateLossTime      int64                       `json:"last_state_loss_time,omitempty" url:"last_state_loss_time,omitempty"`
+	LastActivityTime       int64                       `json:"last_activity_time,omitempty" url:"last_activity_time,omitempty"`
+	ClusterMemoryMb        int64                       `json:"cluster_memory_mb,omitempty" url:"cluster_memory_mb,omitempty"`
+	ClusterCores           string                      `json:"cluster_cores,omitempty" url:"cluster_cores,omitempty"`
+	DefaultTags            []dbmodels.ClusterTag       `json:"default_tags,omitempty" url:"default_tags,omitempty"`
+	ClusterLogStatus       *dbmodels.LogSyncStatus     `json:"cluster_log_status,omitempty" url:"cluster_log_status,omitempty"`
+	TerminationReason      *dbmodels.TerminationReason `json:"termination_reason,omitempty" url:"termination_reason,omitempty"`
+}
+
+func (i *DclusterInfo) FromDataBricksClusterInfo(ci dbmodels.ClusterInfo) *DclusterInfo {
```
Comment (Azadehkhojandi) [inline api/v1/dcluster_types_extra.go]: rename "i" to "dci" to have consistent naming convention

Comment (xinsnake) [reply api/v1/dcluster_types_extra.go]: Renamed the variable name as suggested. Also added reason why this method is needed.

**Pull request merged**